src/EditorFeatures/Core.Wpf/ExternalAccess/VSTypeScript/VSTypeScriptSignatureHelpClassifierProvider.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.Editor.Implementation.IntelliSense.SignatureHelp.Presentation;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.ExternalAccess.VSTypeScript.Api;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Classification;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.VSTypeScript
+{
+    [Export(typeof(IVSTypeScriptSignatureHelpClassifierProvider))]
```

Comment:
@uniqueiniquity so the code that currently uses the signature help classifier in TypeScript, is that just not being used on the Cocoa side at all?

Reply:
Included in EditorFeatures.Cocoa via wildcard.

https://github.com/dotnet/roslyn/blob/36d4d6ee9419e3b7d1224e10c13f2559820a50f8/src/EditorFeatures/Core.Cocoa/Microsoft.CodeAnalysis.EditorFeatures.Cocoa.csproj#L13

Reply:
I don't think we use this type at all...
Here's our `IClassifierProvider`: https://dev.azure.com/devdiv/DevDiv/_git/TypeScript-VS?path=%2FVS%2FLanguageService%2FTypeScriptLanguageService%2FFeatures%2FSignatureHelp%2FTypeScriptSignatureHelpClassifierProvider.cs

Reply:
Oh duh you added this. My mistake.

Reply:
Currently the code uses a `static class` from the Roslyn side. I had to change this from a static factory to a MEF export since it in turn needs access to another export. TypeScript will need to update this reference during the update; everything else should automatically re-bind to the definitions in the new layer.

Reply:
i.e. Instead of calling `VSTypeScriptSignatureHelpClassifierFactory.Create`, you'll have to import `IVSTypeScriptSignatureHelpClassifierProvider` and call its `Create` method. Since `TypeScriptSignatureHelpClassifierProvider` is already a MEF part, this is a trivial change.